utils/firestore.js

```diff
@@ -534,27 +534,30 @@ const removeUninvitedParticipants = async () => {
     }
 }
 
+/**
+ * Get site codes of children entities
+ * @param {string} id - Entity ID
+ */
 const getChildren = async (id) => {
     try{
         const snapShot = await db.collection('siteDetails')
                                 .where('state.parentID', 'array-contains', id)
                                 .get();
         if(snapShot.size > 0) {
+            /** @type {number[]} */
             const siteCodes = [];
-            snapShot.docs.map(document => {
+            snapShot.docs.forEach(document => {
                 if(document.data().siteCode){
                     siteCodes.push(document.data().siteCode);
                 }
             });
             return siteCodes;
         }
-        else{
-            return false;
-        };
+        return [];
     }
     catch(error){
         console.error(error);
-        return new Error(error);
+        return [];
```

Comment:
Question: Does this pattern of not throwing an error still log in DataDog or somewhere on the backend? I'm just asking so I understand how we trace a silent error like this. I do see the conditional handling in `isParentEntity()`...maybe that's my answer?

Reply:
I prefer this patten, so the returned data have consistent types/shapes. When you re-throw the error, it still need to be handled, which usually means printed, upper in the call stack. Yes, the error message, if occurred in above code, will be printed in backend and all these logs are exported to DataDog. Since we can identify error messages from logs, we may not need to take the efforts to pass error messages further up.